generators/python-v2/codegen/src/ast/core/AstNode.ts

```diff
@@ -14,6 +14,12 @@ export abstract class AstNode extends AbstractAstNode {
         return this.references;
     }
 
+    public inheritReferences(astNode: AstNode): void {
+        astNode.getReferences().forEach((reference) => {
+            this.addReference(reference);
+        });
+    }
```

Comment:
when would something have to inherit references? (asking to better understand the use case)

Reply:
"Inherit" might not be the right term since it's actually the reverse direction, but the idea is that if a "parent" AstNode has a "child" AstNode, and that child has a reference, then the parent should have the reference too, such that if you call `write` on the parent, it includes the imports needed by the child.

A real world example for us:

```python
chat_message = ChatMessage(role="USER", content=ChatMessageContent(type="FUNCTION_CALL", value=ChatMessageContentFunctionCall(...)))
```

now imagine our codegen looks like:
```
public ChatMessage extends AstNode {...}

public ChatMessageContent extends AstNode {...}

public ChatMessageContentFunctionCall extends AstNode {...}

chat_message.write(writer)
```

Calling `.write` on `ChatMessage` should include the imports of the children within. One thing to keep in mind is that as of [this PR](https://github.com/fern-api/fern/pull/4948), references are tracked on ast nodes rather than the writer, so having parent nodes inherit references from children nodes is now important.

Reply:
makes sense, i think we should model this as `protected` / `overriden` methods (ie your AST node should call super() and then add its own thing)

Is that something you'd be up to do?

Reply:
https://github.com/fern-api/fern/blob/bfc7aa0efe6800c7a6f1cdfa908778d68d954c98/generators/python-v2/codegen/src/ast/core/AstNode.ts#L7 TLDR lets make this `protected`

Reply:
and then all descendants can modify it how they like and when they like